src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -1321,6 +1345,135 @@ static class B
             Assert.Null(symbolInfo.Symbol);
         }
 
+        [Fact]
+        public void MethodGroup_ExtensionMethodsSameScope_1()
+        {
+            var source = """
+new Program().M();
+
+partial class Program
+{
+    public void M()
+    {
+        System.Delegate d = this.F;
+        System.Console.Write("{0}: {1}", d.GetDelegateMethodName(), d.GetDelegateTypeName());
+    }
+}
+static class A
+{
+    internal static void F(this object x, object y) { }
+}
+static class B
+{
+    internal static void F<T>(this object x, T y) { }
+}
+""";
+            var comp = CreateCompilation(new[] { source, s_utils }, parseOptions: TestOptions.RegularPreview, options: TestOptions.ReleaseExe);
+            comp.VerifyDiagnostics();
```

Comment:
>.VerifyDiagnostics(); [](http://example.com/codeflow?start=16&length=21)

Consider verifying diagnostics on result of ```CompileAndVerify``` instead. Applicable to other added tests as well.